hare/broker.go

```diff
@@ -308,9 +309,12 @@ func (b *Broker) Register(id instanceID) (chan *Msg, error) {
 		b.updateLatestLayer(id)
 
 		if len(b.outbox) >= b.limit {
-			resErr <- errTooMany
-			resCh <- nil
-			return
+			//unregister the earliest layer to make space for the new layer
+			//cannot call unregister here because unregister blocks and this would cause a deadlock
+			id := b.minDeleted + 1
+			delete(b.outbox, id)
```

Comment:
Don't duplicate the code from the `Unregister` method. Refactor that method to create a new (internal) method that does not block, that you can call here directly.